src/Oscoin/CLI/KeyStore.hs

```diff
@@ -61,6 +63,21 @@ ensureConfigDir mbUserPath = do
     configDir <- getConfigPath mbUserPath
     liftIO $ createDirectoryIfMissing True configDir
 
+data KeyStoreError =
+    PrivateKeyDeserialisationFailure DeserialiseFailure
+
+instance Show KeyStoreError where
```

Comment:
nit: In the spirit of `read . show = id`, shouldn't this be just a derived `Show`, and `displayException` showing the descriptive error?

Reply:
That would be my preference as well. However, there were some turbo-shenanigans when I've tried that (i.e. it seemed like the exception was being shown via `show` rather than `displayException`). But this deserves a more accurate try this morning, with a semi-working brain 😁

Reply:
Alright, unless my reptilian brain is deceiving me, I indeed tried again the above, and given this code:

```haskell
data KeyStoreError =
    PrivateKeyDeserialisationFailure DeserialiseFailure

instance Show KeyStoreError where
    show = \case
      PrivateKeyDeserialisationFailure cborError ->
          mconcat [
                    "Loading the oscoin's private key from the configuration "
                  , "folder failed. The reason was: "
                  , show cborError
                  ]

instance Exception KeyStoreError where
    displayException _ = "TODO"
```

It doesn't work as expected -- the exception is shown via `show` it seems, rather than `displayException`. (i.e. I don't see "TODO"). If I can speculate, As `displayException` has been added fairly recently (I think from base 4.8 onwards if I am not mistaken) the exception handler installed in the main thread doesn't display the exception via `displayException` but rather via `show`. I can surely adjust things to behave as we want, if we are OK with the slightly more invasive patch.

Reply:
meh, dissapointing